rest/blip_sync.go

```diff
@@ -603,21 +603,21 @@ func (bh *blipHandler) handleProposedChanges(rq *blip.Message) error {
 func (bh *blipHandler) sendRevAsDelta(sender *blip.Sender, seq db.SequenceID, docID string, revID string, knownRevs map[string]bool, maxHistory int) {
 
 	bh.db.DbStats.StatsDeltaSync().Add(base.StatKeyDeltasRequested, 1)
-	bh.Logf(base.LevelTrace, base.KeySync, "DELTA: getting newBody docID: %s - rev: %v", base.UD(docID), revID)
+	// TODO: determine whether we need to fetch the current rev in order to identify the 'from' rev for the delta,
+	//      or can skip straight to GetDelta with knownRevs[0]
 	newBody, err := bh.db.GetRev(docID, revID, true, nil)
 	if err != nil {
-		bh.Logf(base.LevelWarn, base.KeySync, "DELTA: couldn't get newBody - sending NoRev... err: %v", err)
+		bh.Logf(base.LevelDebug, base.KeySync, "DELTA: couldn't get newBody in order to find common ancestor - sending NoRev... err: %v", err)
 		// If we can't get the new doc, we won't be able to fall back either, so NoRev with error
 		bh.sendNoRev(err, sender, seq, docID, revID)
 		return
 	}
 
 	// Get the client's highest known rev ID to use as deltaSrc
-	bh.Logf(base.LevelTrace, base.KeySync, "DELTA: Figuring out deltaSrcRevID from client's highest known revID...")
 	var deltaSrcRevID string
 	newBodyRevisions, ok := newBody[db.BodyRevisions].(db.Revisions)
 	if !ok {
-		bh.Logf(base.LevelWarn, base.KeySync, "DELTA: couldn't get newBody[_revisions] - sending NoRev... err: %v", err)
+		bh.Logf(base.LevelDebug, base.KeySync, "DELTA: couldn't get newBody[_revisions] - sending NoRev... err: %v", err)
```

Comment:
It looks like this log is actually inaccurate, since we fall back to sending the full body.

```suggestion
		bh.Logf(base.LevelDebug, base.KeySync, "DELTA: couldn't get newBody[_revisions] in order to find common ancestor - sending full-body rev")
```